### PR TITLE
Dummy under fs

### PR DESCRIPTION
**Pull request by ShaoshanLiu**
creating a smart Dummy UnderFS which includes Dummy In and Out Streams.  Calvin, please take a look and let me know if you see any problems.  I will also be running tests locally to verify that it works fine.

Thanks!

Shaoshan

### CODE REVIEW

File: core/src/main/java/tachyon/DummyFileInputStream.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import java.io.InputStream;
+
+public class DummyFileInputStream extends InputStream {
+
+  public DummyFileInputStream() {
+  }
+
+  public int read() {
+    return 0;
```
Comment (calvinjia) [inline core/src/main/java/tachyon/DummyFileInputStream.java]: This might cause infinite loops?

Comment (ShaoshanLiu) [inline core/src/main/java/tachyon/DummyFileInputStream.java]: good catch

**Comment by calvinjia** [comment]
@ShaoshanLiu Are the tests you are doing integration tests? You can try adding the DummyFS as the ufs option in a new test profile if you want to leverage the existing unit tests.

**Comment by ShaoshanLiu** [comment]
I am testing it in a 100-node setup environment with our workloads, I agree, I will add a new test profile to test this as well.  Do you want to merge this into the current branch first, then I can add the test in another PR?

**Comment by calvinjia** [comment]
+1 Sounds good to me.

**Pull request merged**